### ISSUE TITLE
Type unsafe comparison triggers on comparable objects

### ISSUE THREAD

**Issue by Rarst** [kalessil/phpinspectionsea#1109]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive                          |
| Plugin         | Php Inspections (EA Extended) 3.0.8.1 |
| Language level | PHP 7.2                         |

Current behaviour (description/screenshot:)
--

```php
$today1 = date_create('today');
$today2 = date_create('today');
var_dump($today1 == $today2); // Please consider using more strict '===' here
// true, intended
var_dump($today1 === $today2); // No warning
// false, unintended
```

Expected behaviour
--
It would make sense for this inspection to account for comparable objects.

I found the earlier #19 issue about this, but I am not sure what the expected behavior was after the fix and is current one a regression.

Environment details
---
PhpStorm 2018.2.5
Build #PS-182.4892.16, built on October 11, 2018
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 7 6.1
